### PR TITLE
Feature/orlicht/add customization new api version v1

### PR DESCRIPTION
**Pull request by ori-licht**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. - 5.11.2021
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet - 5.11.2021
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by liemilyg — APPROVED** [review]

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntitiesCommon.json [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+          {
+            "description": "Entity represents submission mail in the system.",
+            "value": "SubmissionMail"
+          }
+        ]
+      }
+    }
+  },
+  "parameters": {
+
```
Comment (dosegal) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntitiesCommon.json:191]: extra line - remove

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntitiesCommon.json [hunk 2]
```diff
@@ -0,0 +1,193 @@
+{
```
Comment (dosegal) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntitiesCommon.json:1]: maybe "EntitiesCommonDefenitions"?

Comment (ori-licht) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntitiesCommon.json:1]: mmm..
I like this suggestion, thank you

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntityQueryTemplates.json [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        "x-ms-examples": {
+          "Get all entity query templates.": {
+            "$ref": "./examples/entityQueryTemplates/GetEntityQueryTemplates.json"
+          }
+        },
+        "tags": [
+          "EntityQuery Templates"
+        ],
+        "description": "Gets all entity query templates.",
+        "operationId": "entityQueryTemplates_List",
```
Comment (dosegal) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntityQueryTemplates.json:48]: Tools and libraries use the operationId to identify the operation and the function names, so please follow the naming conventions and start with a capital letter.

Comment (ori-licht) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntityQueryTemplates.json:48]: great catch, this is a mistake, thank you

File: specification/securityinsights/resource-manager/readme.md [hunk 4]
```diff
@@ -53,11 +53,21 @@ These settings apply only when `--tag=package-2021-03-preview-only` is specified
...
 - Microsoft.SecurityInsights/preview/2021-03-01-preview/operations.json
+- Microsoft.SecurityInsights/preview/2021-03-01-preview/EntityQueryTemplates.json
+- Microsoft.SecurityInsights/preview/2021-03-01-preview/EntityQueries.json
 directive:
   - suppress: R4017
     from: Microsoft.SecurityInsights/preview/2021-03-01-preview/Settings.json
     where: $.definitions.Settings
-    reason: The Setting does not support list by subscription. It's not a top-level resource. To get the Watchlist, we should have a subscription as well as a resource group and Log Analytics workspace. 
+    reason: The Setting does not support list by subscription. It's not a top-level resource. To get the Watchlist, we should have a subscription as well as a resource group and Log Analytics workspace.
+  - suppress: R4017
```
Comment (ofshomro) [inline specification/securityinsights/resource-manager/readme.md:63]: Do we need the suppress for every file? can we mention it once for all resources?

Comment (ori-licht) [reply specification/securityinsights/resource-manager/readme.md:63]: this is a good question; I'm relying on @yuvalshilo implementation

**Comment by ori-licht** [comment]
please don't complete this PR, there is still work in progress

**Comment by chiragg4u** [comment]
Please go through the contribution checklist for ARM API Review added as part of the review and mark them completed. It will help you fix most of the issues and improve the quality of swagger and help close the PR faster. This is a requirement before review can be continued.

**Comment by ori-licht** [comment]
> Please go through the contribution checklist for ARM API Review added as part of the review and mark them completed. It will help you fix most of the issues and improve the quality of swagger and help close the PR faster. This is a requirement before review can be continued.

Thank you @chiragg4u,
Can you please point me to which issues you are referring to?
I have invoked the oav for both spec and examples - had no errors
I have invoked the autorest and I only got this: 
image

which should be resolved because the securityInsight is under a nested resource for LA workspace

**Comment by mentat9** [comment]
Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

**Comment by mentat9** [comment]
I worked on reviewing the API changes in this PR over the weekend, but there are so many refactoring changes, it's too hard to see what's changed in the API. Please split this PR into two separate PRs. The first would be just the API changes without reordering, renaming, file renames, file moves, etc. The second would be all of the refactoring changes. When you create the second PR, identify in the comments that you aren't changing the API, just refacoring the swagger.

**IMPORTANT**: for API review always follow the guidelines here: https://armwiki.azurewebsites.net/rp_onboarding/process/api_review.html.

**Comment by ori-licht** [comment]
> I worked on reviewing the API changes in this PR over the weekend, but there are so many refactoring changes, it's too hard to see what's changed in the API. Please split this PR into two separate PRs. The first would be just the API changes without reordering, renaming, file renames, file moves, etc. The second would be all of the refactoring changes. When you create the second PR, identify in the comments that you aren't changing the API, just refacoring the swagger.
> 
> **IMPORTANT**: for API review always follow the guidelines here: https://armwiki.azurewebsites.net/rp_onboarding/process/api_review.html.

@mentat9 - thank for reviewing
our team (Security Insights) is currently working on splitting our swagger - meaning breaking down the SecurityInsights.json to several <resourceType>.json.

this PR represents the following:
1. creating a new api version for the existing entityQueries resource, currently supporting get and getAll 
2. adding put and delete operations to the existing entityQueries resource with the supported entityQueryKind for these requests ("Activity") for put operations.
3. revealing a new resource type, entityQueryTemplates for supporting get and getAll operations and the ActivityEntityQueryTemplate that supports this new resource type.
4. entities definition for common definitions

creating a PR that copies the existing API will contradict our motion to split by resource type.
so, I'm not sure how you suggest I should do it.

I agree this PR is big, we can first create a PR that shows the changes in the entityQuery resource for supporting the new put and delete operations.
then I can create another PR for introducing the new resource (entityQueryTemplates).
WDYT?

**Pull request closed without merging**